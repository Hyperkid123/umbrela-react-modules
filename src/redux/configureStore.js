import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
