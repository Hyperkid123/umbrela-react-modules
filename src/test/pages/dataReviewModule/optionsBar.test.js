import React from 'react'
import OptionsBar, {OptionsBar as Snapshot} from '../../../pages/dataReviewModule/optionsBar';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('Research editor landing component', () => {
  it('should render correclty' , () => {
      const tree = shallow(
        <Snapshot translate={jest.fn()}/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  const mockStore = configureStore();
  let initialState = {
    ui: {
      hideSheets: false,
      hideQuestions: false,
      hideChartlegend: false,
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<OptionsBar store={store}/>);
    expect(wrapper).toBeDefined();
  })

  it('should handle menu open correclty' , () => {
      const wrapper = shallow(
        <Snapshot translate={jest.fn()}/>
      );
      wrapper.instance().handleMenuOpen({currentTarget: 'foo'});
      expect(wrapper.state().menuAnchor).toEqual('foo')
  });

  it('should handle menu close correctly', () => {
    const wrapper = shallow(
      <Snapshot translate={jest.fn()}/>
    );
    wrapper.instance().handleMenuClose();
    expect(wrapper.state().menuAnchor).toEqual(null)
  })

  it('should mount correctly', () => {
    const changeChartType = jest.fn();
    const wrapper = mount(
      <Snapshot changeChartType={changeChartType} translate={jest.fn()}/>
    );
    expect(wrapper).toBeDefined()
  })
});
