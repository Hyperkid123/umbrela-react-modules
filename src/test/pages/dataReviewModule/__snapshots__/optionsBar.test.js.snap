// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Research editor landing component should render correclty 1`] = `
<div
  className="sc-bdVaJa fizmbb"
>
  <WithStyles(Button)>
    <pure(ChevronLeft) />
  </WithStyles(Button)>
  <WithStyles(Button)>
    <pure(ChevronLeft) />
  </WithStyles(Button)>
  <div
    style={
      Object {
        "marginLeft": "auto",
      }
    }
  >
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Switch)
          checked={undefined}
          onChange={undefined}
        />
      }
    />
  </div>
  <WithStyles(Button)
    onClick={[Function]}
  >
    <pure(InsertChart) />
  </WithStyles(Button)>
  <WithStyles(Menu)
    anchorEl={null}
    onClose={[Function]}
    open={false}
    transition={[Function]}
  >
    <WithStyles(MenuItem)
      onClick={[Function]}
    />
    <WithStyles(MenuItem)
      onClick={[Function]}
    />
    <WithStyles(MenuItem)
      onClick={[Function]}
    />
    <WithStyles(MenuItem)
      onClick={[Function]}
    />
    <WithStyles(MenuItem)
      onClick={[Function]}
    />
  </WithStyles(Menu)>
</div>
`;
